### PR TITLE
[APPC-3186] Hide CVC input for a stored card

### PR DESCRIPTION
**Pull request by gustavoaptoide**
**What does this PR do?**

   Hide CVC input for a Stored Card inside payment and TopUp

**Database changed?**

   No

**How should this be manually tested?**

  Try to make a payment and topUp with stored Card

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-3186](https://aptoide.atlassian.net/browse/APPC-3186)

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] Database migration (if applicable)?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

[APPC-3186]: https://aptoide.atlassian.net/browse/APPC-3186?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt [hunk 1]
```diff
@@ -354,12 +366,40 @@ class AdyenPaymentFragment : BasePageViewFragment(), AdyenPaymentView {
...
     setStoredPaymentInformation(isStored)
   }
 
+  private fun setStoredCardLayoutValues(storedPaymentMethod: StoredPaymentMethod) {
+    txt_stored_card_number?.text = "**** ".plus(storedPaymentMethod.lastFour)
+    txt_stored_expiry_date?.text =
+      getString(R.string.dialog_expiry_date).plus(" ").plus(storedPaymentMethod.expiryMonth)
+        .plus("/").plus(storedPaymentMethod.expiryYear)
+    when (storedPaymentMethod.brand) {
+      "mc" -> {
```
[the remark this answers is not in the document]
Comment (gustavoaptoide) [reply app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt:382]: this value comes from adyen so I can't change it to enum

File: app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt [hunk 2]
```diff
@@ -582,6 +624,11 @@ class AdyenPaymentFragment : BasePageViewFragment(), AdyenPaymentView {
     else RxView.clicks(change_card_button_pre_selected!!)
   }
 
+  override fun forgetStoredCardClick(): Observable<Any> {
+    return if (btn_stored_card_pre_selected_change_card != null) RxView.clicks(btn_stored_card_pre_selected_change_card!!)
+    else RxView.clicks(btn_stored_card_change_card!!)
```
[the remark this answers is not in the document]
Comment (gustavoaptoide) [reply app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt:629]: We need use !! beacause the function needs an Observable<Any>

File: app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (giovanni-aptoide) [reply app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt:382]: I'm talking about map with enum, something like that
enum class PaymentBrand(name: String) {
    MC("mc"), 
    VISA("visa"), 
    AMEX("amex"), 
    MAESTRO("maestro"), 
    DINERS("diners"), 
    DISCOVER("discover")
  }

File: app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt [hunk 3]
```diff
@@ -682,8 +735,23 @@ class AdyenPaymentFragment : BasePageViewFragment(), AdyenPaymentView {
...
+    if (needCVC && isStored) {
+      change_card_button?.visibility = VISIBLE
+      change_card_button_pre_selected?.visibility = VISIBLE
+    } else if (isStored) {
+      layout_adyen_stored_card?.visibility = VISIBLE
+      change_card_button_pre_selected?.visibility = GONE
+      layout_pre_selected?.visibility = GONE
+      adyen_card_form?.visibility = GONE
+      change_card_button?.visibility = GONE
+      more_payment_methods?.visibility = VISIBLE
```
[the remark this answers is not in the document]
Comment (gustavoaptoide) [reply app/src/main/java/com/asfoundation/wallet/billing/adyen/AdyenPaymentFragment.kt:746]: I know it's more verbose to put this code, but it's not better for whoever reads it to really understand which layouts are hidden. than seeing just one line and not realizing that there will be several hidden layouts

**Review by giovanni-aptoide — DISMISSED** [review]

**Review by rodolfobcardoso — DISMISSED** [review]

**Review by giovanni-aptoide — APPROVED** [review]

**Pull request merged**